frontend/src/components/contracts/ContractDetails.tsx

```diff
@@ -0,0 +1,130 @@
+import moment from "moment";
+
+import React from "react";
+
+import { Row, Col } from "react-bootstrap";
+
+import ContractsApi from "../../libraries/explorer-wamp/contracts";
+
+import CardCell from "../utils/CardCell";
+import TransactionLink from "../utils/TransactionLink";
+
+interface Props {
+  accountId: string;
+}
+
+interface State {
+  locked?: boolean;
+  transactionHash?: string;
+  timestamp?: number;
+}
+
+export default class extends React.Component<Props, State> {
+  state: State = {};
+
+  collectContractInfo = async () => {
+    new ContractsApi()
+      .getContractInfo(this.props.accountId)
+      .then(contractInfo => {
+        if (contractInfo) {
+          this.setState({
+            transactionHash: contractInfo.transactionHash,
+            timestamp: contractInfo.timestamp,
+            locked: contractInfo.accessKeys.every(
+              (key: any) =>
+                key["access_key"]["permission"]["FunctionCall"] !== undefined
+            )
+          });
+        }
+        return;
+      })
+      .catch(err => console.error(err));
+  };
+
+  componentDidMount() {
+    this.collectContractInfo();
+  }
+
+  componentDidUpdate(preProps: Props) {
+    if (this.props.accountId !== preProps.accountId) {
+      this.collectContractInfo();
+    }
+  }
+  render() {
+    const { locked, transactionHash, timestamp } = this.state;
+    let lockedShow;
+    if (locked !== undefined) {
+      lockedShow = locked === true ? "Locked" : "UnLocked";
+    }
+    return transactionHash ? (
+      <>
+        <div className="contract-title">
+          <img
+            src={"/static/images/icon-d-contract.svg"}
+            className="card-cell-title-img"
+          />
+          CONTRACT
+        </div>
+        <div className="contract-info-container">
+          <Row noGutters className="border-0">
+            <Col md="6">
+              <CardCell
+                title="Code Hash"
+                text={
+                  <TransactionLink transactionHash={transactionHash}>
+                    {transactionHash}
```

Comment:
Code hash is the checksum of the Wasm binary blob, that is returned with the `query(account/<account_id>)` API.

The transaction hash when the code was last updated should be a separate thing if needed /cc @kcole16

Reply:
if that, wait for new design